events/code_review_events/workflow.py

```diff
@@ -216,9 +257,140 @@ def publish_results(self, payload):
+
+        # If this diff does not belong to a revision we pushed to try, return.
+        if diff_id not in self.diff_to_push:
+            logger.warning(
+                "bugbug test select notification for a revision we did not push to try",
+                payload=payload,
+            )
+            return
+
+        # Retrieve artifacts from bugbug test selection task.
+        try:
+            failure_risk = self.community_tc["queue"].getLatestArtifact(
+                bugbug_task_id, "public/failure_risk"
+            )
+            assert isinstance(failure_risk, int)
+        except taskcluster.exceptions.TaskclusterRestFailure as e:
+            if e.status_code == 404:
+                logger.error(
+                    f"Missing failure_risk artifact in bugbug task", task=bugbug_task_id
+                )
+                return
+            raise
+
+        # If the failure risk is low, don't trigger tests.
+        if failure_risk == 0:
+            return
+
+        try:
+            selected_tasks = self.community_tc["queue"].getLatestArtifact(
+                bugbug_task_id, "public/selected_tasks"
+            )
+            # XXX: For now, only restrict to test-linux64 tasks.
+            selected_tasks = {
+                "tasks": [
+                    t
+                    for t in selected_tasks["response"].text.splitlines()
+                    if t.startswith("test-linux64/")
+                ]
+            }
+            if len(selected_tasks["tasks"]) == 0:
+                return
+        except taskcluster.exceptions.TaskclusterRestFailure as e:
+            if e.status_code == 404:
+                logger.error(
+                    f"Missing selected_tasks artifact in bugbug task",
+                    task=bugbug_task_id,
+                )
+                return
+            raise
+
+        # Get the decision task of the push to try.
+        push = self.diff_to_push.pop(diff_id)
+        decision_task_index = self.index_service.findTask(
+            "gecko.v2.try.revision.{}.taskgraph.decision".format(push["revision"])
+        )
+        decision_task_id = decision_task_index["taskId"]
+
+        # Find the 'add-new-jobs' action to add new jobs to the task group.
+        actions = self.queue_service.getLatestArtifact(
+            decision_task_id, "public/actions.json"
+        )
+        add_job_action = [
+            action for action in actions["actions"] if action["name"] == "add-new-jobs"
+        ][0]
+        assert add_job_action["kind"] == "hook"
+
+        # Trigger the 'add-new-jobs' action with the list of tasks bugbug selected.
+        jsonschema.validate(instance=selected_tasks, schema=add_job_action["schema"])
+
+        hookPayload = jsone.render(
+            add_job_action["hookPayload"],
+            context={
+                "taskId": None,
+                "taskGroupId": decision_task_id,
+                "input": selected_tasks,
+            },
+        )
```

Comment:
I would extract all that rendering stuff in a function to be able to unit test it

Reply:
I've extracted it in a separate function to wrap it in try-catch, but I haven't added more tests. Since both the input and the schema to the rendering/validation stuff come from Taskcluster, we don't have much to test ourselves.